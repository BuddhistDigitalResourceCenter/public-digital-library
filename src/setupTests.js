import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import { XMLHttpRequest } from 'xmlhttprequest';
global.XMLHttpRequest = XMLHttpRequest;

jasmine.DEFAULT_TIMEOUT_INTERVAL = 100000;

configure({ adapter: new Adapter() });

var localStorageMock = (function() {
  var store = {};
  return {
    getItem: function(key) {
      return store[key];
    },
    setItem: function(key, value) {
      store[key] = value.toString();
    },
    clear: function() {
      store = {};
    },
    removeItem: function(key) {
      delete store[key];
    }
  };
})();

Object.defineProperty(window, 'localStorage', { value: localStorageMock });

// debug mode VS quiet output
console.log = () => {}

/*
// deprecated - use json-server instead
Object.defineProperty(window, 'fetch', { value: function(url,options,param)
   {
      console.log("fetching",url,options,param);
   }
});
*/
