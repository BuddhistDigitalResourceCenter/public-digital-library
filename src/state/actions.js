// @flow

export type Action = {
    type: string,
    payload?: any,
    error?: Error,
    meta?: any
}

export const INITIATE_APP = 'INITIATE_APP';
export const initiateApp = (params:{},iri?:string,auth?:{},route?:string): Action => {
    return {
        type: INITIATE_APP,
        payload:params,
        meta:{iri,auth,route}
    }
}
