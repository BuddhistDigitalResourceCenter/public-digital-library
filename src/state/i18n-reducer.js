import {
  I18NEXT_INIT,
  I18NEXT_READY,
  I18NEXT_CHANGE_LANGUAGE,
  I18NEXT_CHANGE_LANGUAGE_READY
} from 'i18next-redux-saga';


const initState = {
  loading: 0,
  lang: 'en',
};

export default function appReducer(state = initState, action) {
  const { type, payload } = action;
  switch (type) {
    case I18NEXT_INIT:
      return {
        ...state,
        loading: state.loading + 1,
      };
    case I18NEXT_READY:
    case I18NEXT_CHANGE_LANGUAGE_READY:
      return {
        ...state,
        loading: state.loading - 1,
      };
    case I18NEXT_CHANGE_LANGUAGE:
      return {
        ...state,
        loading: state.loading + 1,
        lang: payload || 'en',
      };
    default:
      return state;
  }
}
