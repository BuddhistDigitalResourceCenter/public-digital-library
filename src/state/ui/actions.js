// @flow
import { createAction } from 'redux-actions';
import type { Action } from '../actions';

export const TYPES = {};

TYPES.helloWorld = 'HELLO_WORLD';
export const helloWorld = createAction(TYPES.helloWorld);

TYPES.toggleLanguagePanel = 'TOGGLE_LANGUAGE_PANEL';
export const toggleLanguagePanel = createAction(TYPES.toggleLanguagePanel);

TYPES.toggleCollapse = 'TOGGLE_COLLAPSE';
export const toggleCollapse = (txt:string): Action => {
   return {
      type: TYPES.toggleCollapse,
      payload: txt
   }
}

TYPES.showResults = 'SHOW_RESULTS';
export const showResults = (key: string,lang:string): Action => {
   return {
      type: TYPES.showResults,
      payload: { key, lang }
   }
}

TYPES.selectType = 'SELECT_TYPE';
export const selectType = (datatype: string): Action => {
   return {
      type: TYPES.selectType,
      payload: datatype
   }
}


TYPES.logEvent = 'LOG_EVENT';
export const logEvent = (login: boolean): Action => {
   return {
      type: TYPES.logEvent,
      payload: login
   }
}


TYPES.setPrefLang = 'SET_PREFLANG';
export const setPrefLang = (lang: string): Action => {
   return {
      type: TYPES.setPrefLang,
      payload: lang
   }
}

TYPES.loadingGallery = 'LOADING_GALLERY';
export const loadingGallery = (manifest: string): Action => {
   return {
      type: TYPES.loadingGallery,
      payload: manifest
   }
}


export type LoadingAction = {
    type: string,
    payload: {
        keyword: string,
        isLoading: boolean
    }
}


TYPES.loading = 'LOADING';
export const loading = (key: string, isLoading: boolean): LoadingAction => {
    return {
        type: TYPES.loading,
        payload: {
            keyword: key,
            isLoading
        }
    }
}
