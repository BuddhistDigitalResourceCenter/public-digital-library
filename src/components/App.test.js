//@flow
import React from 'react';
import ReactDOM from 'react-dom';
import { shallow,mount } from 'enzyme';
import store from '../index';
import {initiateApp} from '../state/actions';
import tcpPortUsed from 'tcp-port-used'
import SearchBar from 'material-ui-search-bar'

require('../setupTests')
let makeRoutes = require('../routes').default
let bdrcAPI = require('../lib/api').default;

describe('main App tests', () => {

   it('loading config file from json-server', async() => {
      tcpPortUsed.waitUntilUsed(5555).then(async() => {

         const api = new bdrcAPI({server:"http://localhost:5555"});

         let config =  await api.getURLContents("http://localhost:5555/config.json",false);
         expect(config).toMatchSnapshot()

         config =  await api.loadConfig();
         expect(config).toMatchSnapshot()

      })
   })

   it('App initialization', async done => {
      tcpPortUsed.waitUntilUsed(5555).then(async () => {

         const div = document.createElement('div');
         let compo ;
         ReactDOM.render( compo = makeRoutes(), div);
         expect(compo).toMatchSnapshot()

         // inspired from https://stackoverflow.com/questions/51226539/formik-testing-error-the-document-global-was-defined-when-react-was-initializ
         setTimeout( () => { done(); }, 400 );
      })
   })

   /*
   it('start a search', async done => {
      tcpPortUsed.waitUntilUsed(5555).then(async () => {

         let compo = mount(makeRoutes())

         setTimeout( () => {
            compo.find(SearchBar).prop('onRequestSearch')("'od zer")
         }, 1000 );

         // inspired from https://stackoverflow.com/questions/51226539/formik-testing-error-the-document-global-was-defined-when-react-was-initializ
         setTimeout( () => { done(); }, 10000 );
      })
   });
   */

})
