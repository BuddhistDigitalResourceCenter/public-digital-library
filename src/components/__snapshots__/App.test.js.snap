// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`main App tests App initialization 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <MuiThemeProvider
    theme={
      Object {
        "breakpoints": Object {
          "between": [Function],
          "down": [Function],
          "keys": Array [
            "xs",
            "sm",
            "md",
            "lg",
            "xl",
          ],
          "only": [Function],
          "up": [Function],
          "values": Object {
            "lg": 1280,
            "md": 960,
            "sm": 600,
            "xl": 1920,
            "xs": 0,
          },
          "width": [Function],
        },
        "direction": "ltr",
        "mixins": Object {
          "gutters": [Function],
          "toolbar": Object {
            "@media (min-width:0px) and (orientation: landscape)": Object {
              "minHeight": 48,
            },
            "@media (min-width:600px)": Object {
              "minHeight": 64,
            },
            "minHeight": 56,
          },
        },
        "overrides": Object {},
        "palette": Object {
          "action": Object {
            "active": "rgba(0, 0, 0, 0.54)",
            "disabled": "rgba(0, 0, 0, 0.26)",
            "disabledBackground": "rgba(0, 0, 0, 0.12)",
            "hover": "rgba(0, 0, 0, 0.08)",
            "hoverOpacity": 0.08,
            "selected": "rgba(0, 0, 0, 0.14)",
          },
          "augmentColor": [Function],
          "background": Object {
            "default": "#fafafa",
            "paper": "#fff",
          },
          "common": Object {
            "black": "#000",
            "white": "#fff",
          },
          "contrastThreshold": 3,
          "divider": "rgba(0, 0, 0, 0.12)",
          "error": Object {
            "contrastText": "#fff",
            "dark": "#d32f2f",
            "light": "#e57373",
            "main": "#f44336",
          },
          "getContrastText": [Function],
          "grey": Object {
            "100": "#f5f5f5",
            "200": "#eeeeee",
            "300": "#e0e0e0",
            "400": "#bdbdbd",
            "50": "#fafafa",
            "500": "#9e9e9e",
            "600": "#757575",
            "700": "#616161",
            "800": "#424242",
            "900": "#212121",
            "A100": "#d5d5d5",
            "A200": "#aaaaaa",
            "A400": "#303030",
            "A700": "#616161",
          },
          "primary": Object {
            "100": "#c5cae9",
            "200": "#9fa8da",
            "300": "#7986cb",
            "400": "#5c6bc0",
            "50": "#e8eaf6",
            "500": "#3f51b5",
            "600": "#3949ab",
            "700": "#303f9f",
            "800": "#283593",
            "900": "#1a237e",
            "A100": "#8c9eff",
            "A200": "#536dfe",
            "A400": "#3d5afe",
            "A700": "#304ffe",
            "contrastText": "#fff",
            "dark": "#303f9f",
            "light": "#7986cb",
            "main": "#3f51b5",
          },
          "secondary": Object {
            "100": "#c5cae9",
            "200": "#9fa8da",
            "300": "#7986cb",
            "400": "#5c6bc0",
            "50": "#e8eaf6",
            "500": "#3f51b5",
            "600": "#3949ab",
            "700": "#303f9f",
            "800": "#283593",
            "900": "#1a237e",
            "A100": "#8c9eff",
            "A200": "#536dfe",
            "A400": "#3d5afe",
            "A700": "#304ffe",
            "contrastText": "#fff",
            "dark": "#303f9f",
            "light": "#7986cb",
            "main": "#3f51b5",
          },
          "text": Object {
            "disabled": "rgba(0, 0, 0, 0.38)",
            "hint": "rgba(0, 0, 0, 0.38)",
            "primary": "rgba(0, 0, 0, 0.87)",
            "secondary": "rgba(0, 0, 0, 0.54)",
          },
          "tonalOffset": 0.2,
          "type": "light",
        },
        "props": Object {},
        "shadows": Array [
          "none",
          "0px 1px 3px 0px rgba(0, 0, 0, 0.2),0px 1px 1px 0px rgba(0, 0, 0, 0.14),0px 2px 1px -1px rgba(0, 0, 0, 0.12)",
          "0px 1px 5px 0px rgba(0, 0, 0, 0.2),0px 2px 2px 0px rgba(0, 0, 0, 0.14),0px 3px 1px -2px rgba(0, 0, 0, 0.12)",
          "0px 1px 8px 0px rgba(0, 0, 0, 0.2),0px 3px 4px 0px rgba(0, 0, 0, 0.14),0px 3px 3px -2px rgba(0, 0, 0, 0.12)",
          "0px 2px 4px -1px rgba(0, 0, 0, 0.2),0px 4px 5px 0px rgba(0, 0, 0, 0.14),0px 1px 10px 0px rgba(0, 0, 0, 0.12)",
          "0px 3px 5px -1px rgba(0, 0, 0, 0.2),0px 5px 8px 0px rgba(0, 0, 0, 0.14),0px 1px 14px 0px rgba(0, 0, 0, 0.12)",
          "0px 3px 5px -1px rgba(0, 0, 0, 0.2),0px 6px 10px 0px rgba(0, 0, 0, 0.14),0px 1px 18px 0px rgba(0, 0, 0, 0.12)",
          "0px 4px 5px -2px rgba(0, 0, 0, 0.2),0px 7px 10px 1px rgba(0, 0, 0, 0.14),0px 2px 16px 1px rgba(0, 0, 0, 0.12)",
          "0px 5px 5px -3px rgba(0, 0, 0, 0.2),0px 8px 10px 1px rgba(0, 0, 0, 0.14),0px 3px 14px 2px rgba(0, 0, 0, 0.12)",
          "0px 5px 6px -3px rgba(0, 0, 0, 0.2),0px 9px 12px 1px rgba(0, 0, 0, 0.14),0px 3px 16px 2px rgba(0, 0, 0, 0.12)",
          "0px 6px 6px -3px rgba(0, 0, 0, 0.2),0px 10px 14px 1px rgba(0, 0, 0, 0.14),0px 4px 18px 3px rgba(0, 0, 0, 0.12)",
          "0px 6px 7px -4px rgba(0, 0, 0, 0.2),0px 11px 15px 1px rgba(0, 0, 0, 0.14),0px 4px 20px 3px rgba(0, 0, 0, 0.12)",
          "0px 7px 8px -4px rgba(0, 0, 0, 0.2),0px 12px 17px 2px rgba(0, 0, 0, 0.14),0px 5px 22px 4px rgba(0, 0, 0, 0.12)",
          "0px 7px 8px -4px rgba(0, 0, 0, 0.2),0px 13px 19px 2px rgba(0, 0, 0, 0.14),0px 5px 24px 4px rgba(0, 0, 0, 0.12)",
          "0px 7px 9px -4px rgba(0, 0, 0, 0.2),0px 14px 21px 2px rgba(0, 0, 0, 0.14),0px 5px 26px 4px rgba(0, 0, 0, 0.12)",
          "0px 8px 9px -5px rgba(0, 0, 0, 0.2),0px 15px 22px 2px rgba(0, 0, 0, 0.14),0px 6px 28px 5px rgba(0, 0, 0, 0.12)",
          "0px 8px 10px -5px rgba(0, 0, 0, 0.2),0px 16px 24px 2px rgba(0, 0, 0, 0.14),0px 6px 30px 5px rgba(0, 0, 0, 0.12)",
          "0px 8px 11px -5px rgba(0, 0, 0, 0.2),0px 17px 26px 2px rgba(0, 0, 0, 0.14),0px 6px 32px 5px rgba(0, 0, 0, 0.12)",
          "0px 9px 11px -5px rgba(0, 0, 0, 0.2),0px 18px 28px 2px rgba(0, 0, 0, 0.14),0px 7px 34px 6px rgba(0, 0, 0, 0.12)",
          "0px 9px 12px -6px rgba(0, 0, 0, 0.2),0px 19px 29px 2px rgba(0, 0, 0, 0.14),0px 7px 36px 6px rgba(0, 0, 0, 0.12)",
          "0px 10px 13px -6px rgba(0, 0, 0, 0.2),0px 20px 31px 3px rgba(0, 0, 0, 0.14),0px 8px 38px 7px rgba(0, 0, 0, 0.12)",
          "0px 10px 13px -6px rgba(0, 0, 0, 0.2),0px 21px 33px 3px rgba(0, 0, 0, 0.14),0px 8px 40px 7px rgba(0, 0, 0, 0.12)",
          "0px 10px 14px -6px rgba(0, 0, 0, 0.2),0px 22px 35px 3px rgba(0, 0, 0, 0.14),0px 8px 42px 7px rgba(0, 0, 0, 0.12)",
          "0px 11px 14px -7px rgba(0, 0, 0, 0.2),0px 23px 36px 3px rgba(0, 0, 0, 0.14),0px 9px 44px 8px rgba(0, 0, 0, 0.12)",
          "0px 11px 15px -7px rgba(0, 0, 0, 0.2),0px 24px 38px 3px rgba(0, 0, 0, 0.14),0px 9px 46px 8px rgba(0, 0, 0, 0.12)",
        ],
        "shape": Object {
          "borderRadius": 4,
        },
        "spacing": Object {
          "unit": 8,
        },
        "transitions": Object {
          "create": [Function],
          "duration": Object {
            "complex": 375,
            "enteringScreen": 225,
            "leavingScreen": 195,
            "short": 250,
            "shorter": 200,
            "shortest": 150,
            "standard": 300,
          },
          "easing": Object {
            "easeIn": "cubic-bezier(0.4, 0, 1, 1)",
            "easeInOut": "cubic-bezier(0.4, 0, 0.2, 1)",
            "easeOut": "cubic-bezier(0.0, 0, 0.2, 1)",
            "sharp": "cubic-bezier(0.4, 0, 0.6, 1)",
          },
          "getAutoHeightDuration": [Function],
        },
        "typography": Object {
          "body1": Object {
            "color": "rgba(0, 0, 0, 0.87)",
            "fontFamily": "\\"Roboto\\", \\"Helvetica\\", \\"Arial\\", sans-serif",
            "fontSize": "0.875rem",
            "fontWeight": 400,
            "lineHeight": "1.46429em",
          },
          "body2": Object {
            "color": "rgba(0, 0, 0, 0.87)",
            "fontFamily": "\\"Roboto\\", \\"Helvetica\\", \\"Arial\\", sans-serif",
            "fontSize": "0.875rem",
            "fontWeight": 500,
            "lineHeight": "1.71429em",
          },
          "button": Object {
            "color": "rgba(0, 0, 0, 0.87)",
            "fontFamily": "\\"Roboto\\", \\"Helvetica\\", \\"Arial\\", sans-serif",
            "fontSize": "0.875rem",
            "fontWeight": 500,
            "textTransform": "uppercase",
          },
          "caption": Object {
            "color": "rgba(0, 0, 0, 0.54)",
            "fontFamily": "\\"Roboto\\", \\"Helvetica\\", \\"Arial\\", sans-serif",
            "fontSize": "0.75rem",
            "fontWeight": 400,
            "lineHeight": "1.375em",
          },
          "display1": Object {
            "color": "rgba(0, 0, 0, 0.54)",
            "fontFamily": "\\"Roboto\\", \\"Helvetica\\", \\"Arial\\", sans-serif",
            "fontSize": "2.125rem",
            "fontWeight": 400,
            "lineHeight": "1.20588em",
          },
          "display2": Object {
            "color": "rgba(0, 0, 0, 0.54)",
            "fontFamily": "\\"Roboto\\", \\"Helvetica\\", \\"Arial\\", sans-serif",
            "fontSize": "2.8125rem",
            "fontWeight": 400,
            "lineHeight": "1.13333em",
            "marginLeft": "-.02em",
          },
          "display3": Object {
            "color": "rgba(0, 0, 0, 0.54)",
            "fontFamily": "\\"Roboto\\", \\"Helvetica\\", \\"Arial\\", sans-serif",
            "fontSize": "3.5rem",
            "fontWeight": 400,
            "letterSpacing": "-.02em",
            "lineHeight": "1.30357em",
            "marginLeft": "-.02em",
          },
          "display4": Object {
            "color": "rgba(0, 0, 0, 0.54)",
            "fontFamily": "\\"Roboto\\", \\"Helvetica\\", \\"Arial\\", sans-serif",
            "fontSize": "7rem",
            "fontWeight": 300,
            "letterSpacing": "-.04em",
            "lineHeight": "1.14286em",
            "marginLeft": "-.04em",
          },
          "fontFamily": "\\"Roboto\\", \\"Helvetica\\", \\"Arial\\", sans-serif",
          "fontSize": 14,
          "fontWeightLight": 300,
          "fontWeightMedium": 500,
          "fontWeightRegular": 400,
          "headline": Object {
            "color": "rgba(0, 0, 0, 0.87)",
            "fontFamily": "\\"Roboto\\", \\"Helvetica\\", \\"Arial\\", sans-serif",
            "fontSize": "1.5rem",
            "fontWeight": 400,
            "lineHeight": "1.35417em",
          },
          "pxToRem": [Function],
          "round": [Function],
          "subheading": Object {
            "color": "rgba(0, 0, 0, 0.87)",
            "fontFamily": "\\"Roboto\\", \\"Helvetica\\", \\"Arial\\", sans-serif",
            "fontSize": "1rem",
            "fontWeight": 400,
            "lineHeight": "1.5em",
          },
          "title": Object {
            "color": "rgba(0, 0, 0, 0.87)",
            "fontFamily": "\\"Roboto\\", \\"Helvetica\\", \\"Arial\\", sans-serif",
            "fontSize": "1.3125rem",
            "fontWeight": 500,
            "lineHeight": "1.16667em",
          },
        },
        "zIndex": Object {
          "appBar": 1100,
          "drawer": 1200,
          "mobileStepper": 1000,
          "modal": 1300,
          "snackbar": 1400,
          "tooltip": 1500,
        },
      }
    }
  >
    <Router
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "createHref": [Function],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <Switch>
        <Route
          path="/auth/callback"
          render={[Function]}
        />
        <Route
          exact={true}
          path="/"
          render={[Function]}
        />
        <Route
          path="/search"
          render={[Function]}
        />
        <Route
          path="/gallery"
          render={[Function]}
        />
        <Route
          path="/show/bdr::IRI"
          render={[Function]}
        />
        <Route
          render={[Function]}
        />
      </Switch>
    </Router>
  </MuiThemeProvider>
</Provider>
`;

exports[`main App tests App initialization 2`] = `
"{
  \\"auth\\": {
    \\"domain\\": \\"bdrc-io.auth0.com\\",
    \\"clientID\\": \\"bZRCKQvibdCq5gF7gV2biIH9rLcqfVcS\\",
    \\"redirectUri\\": \\"http://localhost:3000/auth/callback\\",
    \\"audience\\": \\"https://bdrc-io.auth0.com/userinfo\\",
    \\"responseType\\": \\"token id_token\\",
    \\"scope\\": \\"openid user_metadata app_metada\\"
  },
  \\"ldspdi\\": {
    \\"endpoints\\": [
      \\"http://localhost:8080\\",
      \\"http://localhost:13280\\",
      \\"http://buda1.bdrc.io:13280\\",
      \\"http://purl.bdrc.io\\",
      \\"http://localhost:5555\\"
    ],
    \\"index\\": 4
  },
  \\"iiif\\": {
    \\"endpoints\\": [
      \\"http://localhost:9000\\",
      \\"http://iiif.bdrc.io\\",
      \\"http://localhost:5555\\"
    ],
    \\"index\\": 2
  },
  \\"rkts\\": {
    \\"endpoints\\": [
      \\"http://localhost:8080\\"
    ],
    \\"index\\": 0
  },
  \\"facets\\": {
    \\"Place\\": {
      \\"relation\\": \\"http://purl.bdrc.io/ontology/tmp/relationType\\"
    },
    \\"Lineage\\": {
      \\"relation\\": \\"http://purl.bdrc.io/ontology/tmp/relationType\\"
    },
    \\"Person\\": {
      \\"relation\\": \\"http://purl.bdrc.io/ontology/tmp/relationType\\",
      \\"gender\\": \\"http://purl.bdrc.io/ontology/core/personGender\\"
    },
    \\"Work\\": {
      \\"relation\\": \\"http://purl.bdrc.io/ontology/tmp/relationType\\",
      \\"langScript\\": \\"http://purl.bdrc.io/ontology/core/workLangScript\\",
      \\"status\\": \\"http://purl.bdrc.io/ontology/admin/status\\",
      \\"access\\": \\"http://purl.bdrc.io/ontology/admin/access\\",
      \\"license\\": \\"http://purl.bdrc.io/ontology/admin/license\\",
      \\"type\\": \\"http://purl.bdrc.io/ontology/core/workType\\",
      \\"partType\\": \\"http://purl.bdrc.io/ontology/core/workPartType\\",
      \\"tree\\": \\"http://purl.bdrc.io/ontology/core/workGenre\\"
    },
    \\"Etext\\": {
      \\"tree\\": \\"http://purl.bdrc.io/ontology/core/workGenre\\"
    }
  },
  \\"links\\": [
    {
      \\"title\\": \\"Abstract text of the Bodhisattvacaryāvatāra\\",
      \\"id\\": \\"W19740\\",
      \\"label\\": \\"spyod 'jug\\",
      \\"lang\\": \\"bo-x-ewts\\",
      \\"icon\\": \\"Abstract Work\\"
    },
    {
      \\"title\\": \\"Shantideva\\",
      \\"id\\": \\"P6161\\",
      \\"label\\": \\"zhi ba lha/\\"
    }
  ]
}"
`;

exports[`main App tests App initialization 3`] = `
Object {
  "auth": Object {
    "audience": "https://bdrc-io.auth0.com/userinfo",
    "clientID": "bZRCKQvibdCq5gF7gV2biIH9rLcqfVcS",
    "domain": "bdrc-io.auth0.com",
    "redirectUri": "http://localhost:3000/auth/callback",
    "responseType": "token id_token",
    "scope": "openid user_metadata app_metada",
  },
  "facets": Object {
    "Etext": Object {
      "tree": "http://purl.bdrc.io/ontology/core/workGenre",
    },
    "Lineage": Object {
      "relation": "http://purl.bdrc.io/ontology/tmp/relationType",
    },
    "Person": Object {
      "gender": "http://purl.bdrc.io/ontology/core/personGender",
      "relation": "http://purl.bdrc.io/ontology/tmp/relationType",
    },
    "Place": Object {
      "relation": "http://purl.bdrc.io/ontology/tmp/relationType",
    },
    "Work": Object {
      "access": "http://purl.bdrc.io/ontology/admin/access",
      "langScript": "http://purl.bdrc.io/ontology/core/workLangScript",
      "license": "http://purl.bdrc.io/ontology/admin/license",
      "partType": "http://purl.bdrc.io/ontology/core/workPartType",
      "relation": "http://purl.bdrc.io/ontology/tmp/relationType",
      "status": "http://purl.bdrc.io/ontology/admin/status",
      "tree": "http://purl.bdrc.io/ontology/core/workGenre",
      "type": "http://purl.bdrc.io/ontology/core/workType",
    },
  },
  "iiif": Object {
    "endpoints": Array [
      "http://localhost:9000",
      "http://iiif.bdrc.io",
      "http://localhost:5555",
    ],
    "index": 2,
  },
  "ldspdi": Object {
    "endpoints": Array [
      "http://localhost:8080",
      "http://localhost:13280",
      "http://buda1.bdrc.io:13280",
      "http://purl.bdrc.io",
      "http://localhost:5555",
    ],
    "index": 4,
  },
  "links": Array [
    Object {
      "icon": "Abstract Work",
      "id": "W19740",
      "label": "spyod 'jug",
      "lang": "bo-x-ewts",
      "title": "Abstract text of the Bodhisattvacaryāvatāra",
    },
    Object {
      "id": "P6161",
      "label": "zhi ba lha/",
      "title": "Shantideva",
    },
  ],
  "rkts": Object {
    "endpoints": Array [
      "http://localhost:8080",
    ],
    "index": 0,
  },
}
`;

exports[`main App tests start a search 1`] = `
Object {
  "auth": Object {
    "audience": "https://bdrc-io.auth0.com/userinfo",
    "clientID": "bZRCKQvibdCq5gF7gV2biIH9rLcqfVcS",
    "domain": "bdrc-io.auth0.com",
    "redirectUri": "http://localhost:3000/auth/callback",
    "responseType": "token id_token",
    "scope": "openid user_metadata app_metada",
  },
  "facets": Object {
    "Etext": Object {
      "tree": "http://purl.bdrc.io/ontology/core/workGenre",
    },
    "Lineage": Object {
      "relation": "http://purl.bdrc.io/ontology/tmp/relationType",
    },
    "Person": Object {
      "gender": "http://purl.bdrc.io/ontology/core/personGender",
      "relation": "http://purl.bdrc.io/ontology/tmp/relationType",
    },
    "Place": Object {
      "relation": "http://purl.bdrc.io/ontology/tmp/relationType",
    },
    "Work": Object {
      "access": "http://purl.bdrc.io/ontology/admin/access",
      "langScript": "http://purl.bdrc.io/ontology/core/workLangScript",
      "license": "http://purl.bdrc.io/ontology/admin/license",
      "partType": "http://purl.bdrc.io/ontology/core/workPartType",
      "relation": "http://purl.bdrc.io/ontology/tmp/relationType",
      "status": "http://purl.bdrc.io/ontology/admin/status",
      "tree": "http://purl.bdrc.io/ontology/core/workGenre",
      "type": "http://purl.bdrc.io/ontology/core/workType",
    },
  },
  "iiif": Object {
    "endpoints": Array [
      "http://localhost:9000",
      "http://iiif.bdrc.io",
      "http://localhost:5555",
    ],
    "index": 2,
  },
  "ldspdi": Object {
    "endpoints": Array [
      "http://localhost:8080",
      "http://localhost:13280",
      "http://buda1.bdrc.io:13280",
      "http://purl.bdrc.io",
      "http://localhost:5555",
    ],
    "index": 4,
  },
  "links": Array [
    Object {
      "icon": "Abstract Work",
      "id": "W19740",
      "label": "spyod 'jug",
      "lang": "bo-x-ewts",
      "title": "Abstract text of the Bodhisattvacaryāvatāra",
    },
    Object {
      "id": "P6161",
      "label": "zhi ba lha/",
      "title": "Shantideva",
    },
  ],
  "rkts": Object {
    "endpoints": Array [
      "http://localhost:8080",
    ],
    "index": 0,
  },
}
`;
